src/java/org/apache/cassandra/locator/AzureSnitch.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.locator;
+
+import java.io.IOException;
+
+import com.google.common.collect.ImmutableMap;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.cassandra.utils.JsonUtils;
+import org.apache.cassandra.utils.Pair;
+
+import static java.lang.String.format;
+
+public class AzureSnitch extends AbstractCloudMetadataServiceSnitch
+{
+    public AzureSnitch() throws IOException
+    {
+        this(new SnitchProperties());
+    }
+
+    public AzureSnitch(SnitchProperties properties) throws IOException
+    {
+        this(new AzureCloudConnector(properties), properties);
+    }
+
+    public AzureSnitch(AzureCloudConnector connector, SnitchProperties properties) throws IOException
+    {
+        super(connector, properties, resolveDcAndRack(connector));
+    }
+
+    private static Pair<String, String> resolveDcAndRack(AzureCloudConnector connector) throws IOException
+    {
+        String response = connector.getResponse();
+        JsonNode jsonNode = JsonUtils.JSON_OBJECT_MAPPER.readTree(response);
+
+        JsonNode location = jsonNode.get("location");
+        JsonNode zone = jsonNode.get("zone");
+
+        String datacenter;
+        String rack;
+
+        if (location == null || location.isNull() || location.asText().isEmpty())
+            datacenter = DEFAULT_DC;
+        else
+            datacenter = location.asText();
```

Comment:
not sure if having multiple data centers in the same location is a goal. If so, this would not work for that scenario.

Reply:
resolved privately, in practice one is supposed to use dc_suffix for this kind of scenario